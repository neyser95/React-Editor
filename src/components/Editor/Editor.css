.EditorLayout{
  display: flex;
  border: solid 1px rgba(0, 0, 0, 0.7);
  width: 100%;
  height: 100vh;
  box-sizing: border-box;
  margin: 10px 0;
}
.Editor{
  width: 100%;
  padding: 10px;
  box-sizing: border-box;
}
.Editor__Split{
  width: 50%;
  padding: 10px;
  box-sizing: border-box;
}

.MarkupEditor{
  border-right: solid 1px rgba(0, 0, 0, 0.7);
  width: 50%;
  padding: 10px;
  box-sizing: border-box;
}

/* * Will only print content within the editor */
@media print {
  body * {
    visibility: hidden;
  }

  .Editor, .Editor * {
    visibility: visible;
  }

  .Editor {
    position: absolute;
    border: none;
    left: 0;
    top: 0;
  }
}