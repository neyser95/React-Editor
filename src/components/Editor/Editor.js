import React from 'react';
import classNames from './Editor.css';

const editor = (props) => {
  return (
    // * Will create a div whose content will be editable.
    <div
      className={classNames.Editor}
      onInput={(e) => props.handleEditorChange(e)} 
      contentEditable>
    </div>
  );
}

export default editor;