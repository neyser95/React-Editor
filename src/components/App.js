import React, { Component } from 'react';
import Toolbar from './Toolbar/Toolbar';
import Editor from './Editor/Editor';

class App extends Component {
  // * Our App will be a class component and hanling all state
  state = {
    // * toolbarButtons will hold the class needed for button icon and also the function the button will invoke when clicked on.
    toolbarButtons: {
      underline: {
        icon: 'fas fa-underline',
        event: function() {
          document.execCommand('underline', false, '');
        }
      },
      italic: {
        icon: 'fas fa-italic',
        event: function() {
          document.execCommand('italic', false, '');
        }
      },
      bold: {
        icon: 'fas fa-bold',
        event: function() {
          document.execCommand('bold', false, '');
        }
      },
      link: {
        icon: 'fas fa-link',
        event: function() {
          const url = prompt('Enter the URL');
          document.execCommand('createLink', false, url);
        }
      },
      cut: {
        icon: 'fas fa-cut',
        event: function() {
          document.execCommand('cut', false, '');
        }
      },
      undo: {
        icon: 'fas fa-undo',
        event: function() {
          document.execCommand('undo', false, '');
        }
      },
      redo: {
        icon: 'fas fa-redo',
        event: function() {
          document.execCommand('redo', false, '');
        }
      },
      color: {
        icon: 'fas fa-tint',
        event: function() {
          const color = prompt('Enter your color in hex ex:#f1f233');
          document.execCommand('foreColor', false, color);
        }
      },
      crossText: {
        icon: 'fas fa-strikethrough',
        event: function() {
          document.execCommand('strikeThrough', false, '');
        }
      },
      delete: {
        icon: 'fas fa-trash-alt',
        event: function() {
          document.execCommand('delete', false, '');
        }
      },
      selectAll: {
        icon: 'fas fa-mouse-pointer',
        event: function() {
          document.execCommand('selectAll', false, '');
        }
      },
      copy: {
        icon: 'fas fa-copy',
        event: function() {
          document.execCommand('copy', false, '');
        }
      },
      alignCenter: {
        icon: 'fas fa-align-center',
        event: function() {
          document.execCommand('justifyCenter', false, '');
        }
      },
      alignLeft: {
        icon: 'fas fa-align-left',
        event: function() {
          document.execCommand('justifyLeft', false, '');
        }
      },
      alignRight: {
        icon: 'fas fa-align-right',
        event: function() {
          document.execCommand('justifyRight', false, '');
        }
      },
      export: {
        icon: 'fas fa-file-export',
        event: ''
      }
    },
    // * This will hold the all text within the editorBody
    editorBody: ''
  };

  // * Will handle all the changes made inside the editor and save it to state for later use.
  handleEditorChange = e => {
    const editorBody = e.target.innerHTML;
    console.log(editorBody);
    this.setState({ editorBody });
  };

  render() {
    return (
      <div>
        <Toolbar toolbarButtons={this.state.toolbarButtons} />
        <Editor handleEditorChange={this.handleEditorChange} />
      </div>
    );
  }
}

export default App;
