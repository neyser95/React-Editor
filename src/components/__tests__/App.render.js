import React from 'react';
import renderer from 'react-test-renderer';
import Editor from '../App';
import App from '../App';

//* Tests to make sure that comopnent does not change unexpectdly.
describe('<App />', () => {
  it('will render with Toolbar and Editor', () => {
    const app = renderer.create(
      <App />
    );

    let appComponent = app.toJSON();
    expect(appComponent).toMatchSnapshot();
  })
});
