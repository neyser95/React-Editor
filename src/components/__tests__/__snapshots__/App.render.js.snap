// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> will render with Toolbar and Editor 1`] = `
<div>
  <div>
    <button
      onClick={[Function]}
    >
      <i
        className="fas fa-underline"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <i
        className="fas fa-italic"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <i
        className="fas fa-bold"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <i
        className="fas fa-link"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <i
        className="fas fa-cut"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <i
        className="fas fa-undo"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <i
        className="fas fa-redo"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <i
        className="fas fa-tint"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <i
        className="fas fa-strikethrough"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <i
        className="fas fa-trash-alt"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <i
        className="fas fa-mouse-pointer"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <i
        className="fas fa-copy"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <i
        className="fas fa-align-center"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <i
        className="fas fa-align-left"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <i
        className="fas fa-align-right"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <i
        className="fab fa-markdown"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <i
        className="fas fa-file-export"
      />
    </button>
  </div>
  <p>
    Please press 
    <i
      class="fab fa-markdown"
    />
     to enable markdown. Press cmd+b for 
    <b>
      Bold
    </b>
    . Press cmd+i for 
    <i>
      Italic
    </i>
    . Press control+u for 
    <u>
      Underline
    </u>
    . Press cmd+a for Select All. Press cmd+p to print.
  </p>
  <div>
    <div
      contentEditable={true}
      onInput={[Function]}
    />
  </div>
</div>
`;
